src/main/java/com/sprint/mission/discodeit/controller/AuthController.java

```diff
@@ -1,26 +1,80 @@
 package com.sprint.mission.discodeit.controller;
 
-import com.sprint.mission.discodeit.controller.api.AuthApi;
-import com.sprint.mission.discodeit.dto.user.AuthRequestDto;
+import com.sprint.mission.discodeit.auth.CustomUserDetails;
+import com.sprint.mission.discodeit.auth.RoleUpdateRequest;
 import com.sprint.mission.discodeit.dto.user.UserDto;
-import com.sprint.mission.discodeit.service.Interface.AuthService;
-import jakarta.validation.Valid;
+import com.sprint.mission.discodeit.entity.User;
+import com.sprint.mission.discodeit.exception.user.UserNotFoundException;
+import com.sprint.mission.discodeit.mapper.UserMapper;
+import com.sprint.mission.discodeit.repository.UserRepository;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.session.SessionInformation;
+import org.springframework.security.core.session.SessionRegistry;
+import org.springframework.security.web.authentication.RememberMeServices;
+import org.springframework.security.web.authentication.rememberme.PersistentTokenBasedRememberMeServices;
+import org.springframework.security.web.csrf.CsrfToken;
 import org.springframework.web.bind.annotation.*;
 
 
 @RestController
 @RequestMapping("/api/auth")
 @RequiredArgsConstructor
-public class AuthController implements AuthApi {
+public class AuthController {
 
-  private final AuthService authService;
+    private final UserMapper userMapper;
+    private final UserRepository userRepository;
+    private final SessionRegistry sessionRegistry;
+    private final RememberMeServices rememberMeServices;
 
-  @Override
-  @PostMapping(value = "/login")
-  public ResponseEntity<UserDto> login(@Valid @RequestBody AuthRequestDto request) {
-    return ResponseEntity.status(HttpStatus.OK).body(authService.login(request));
-  }
+    @GetMapping("/csrf-token")
+    public ResponseEntity<CsrfToken> getCsrfToken(HttpServletRequest request) {
+        CsrfToken csrfToken = (CsrfToken) request.getAttribute(CsrfToken.class.getName());
```

Comment:
인자로 CsrfToken을 바로 선언해도 좋을 것 같아요. 
```suggestion
    public ResponseEntity<CsrfToken> getCsrfToken(CsrfToken csrfToken) {
```